### ISSUE TITLE
thinLTO is linking with one thread?

### ISSUE THREAD

**Issue by nickdesaulniers** [ClangBuiltLinux/linux#1847]
Doing an x86 defconfig build with `CONFIG_LTO_CLANG_THIN=y` enabled, I'm seeing only 1 thread being used by LLD during `LD      vmlinux.o`. Is that expected/a regression?

cc @samitolvanen @MaskRay

**Comment by samitolvanen** [comment]
No, that's not expected. You should see one process with multiple threads. Unless something changed and we now have to explicitly specify the number of ThinLTO jobs?

**Comment by torvic9** [comment]
Maybe related: I've never seen ThinLTO use all of the available CPUs when building the kernel (lld-15 and lld-16). At best, it uses exactly half; in my case, 12 out of 24 on a Zen2.

Is there some sort of a "limit" on how many threads LTO can use?

**Comment by torvic9** [comment]
I checked it again, Linux 6.3.3 + Clang16:

Linking vmlinux.o starts off with using half of the available CPUs, then quickly goes down to just using a single thread.

**Comment by samitolvanen** [comment]
You could try passing `--thinlto-jobs=N` to lld and see if that makes any difference?

https://clang.llvm.org/docs/ThinLTO.html#controlling-backend-parallelism

**Comment by nathanchance** [comment]
Could https://github.com/llvm/llvm-project/commit/a8788de1c3f3c8c3a591bd3aae2acee1b43b229a be related?
